http3/server.go

```diff
@@ -479,6 +497,26 @@ func (s *Server) handleConn(conn quic.Connection) error {
 		}
 	}
 
+	var (
+		// first valid ID is zero, subtract 4, so goaway frame id starts at 0
+		lastID = quic.StreamID(-4)
+		// wait until all request goroutines are done
+		wg sync.WaitGroup
+	)
+
+	s.connCount.Add(1)
```

Comment:
I don't think this is quite right. `sync.WaitGroup` requires you to call `Add` and `Wait` from the same Go routine.

This means that we need to make sure that `handleConn` is not called after the shutdown has been initiated.

Reply:
The document only says `Note that calls with a positive delta that occur when the counter is zero must happen before a Wait.` And deferred function does run in the same goroutine as the surrounding function.

The counter for wait ground is incremented just before creating `handleRequest` goroutine, and that's the correct usage.

Reply:
I'm wondering if we should just make `handleConn` block until all streams have finished, and then do the connection counting logic in the function that calls `handleConn`. What do you think?

Reply:
I guess it can be done.